src/toil/jobStores/abstractJobStore.py

```diff
@@ -495,8 +495,26 @@ def getConnectedJobs(jobGraph):
             # Delete the job
             self.delete(jobGraph.jobStoreID)
 
+        jobGraphsReachableFromRoot = {id: getJob(id) for id in reachableFromRoot}
+
```

Comment:
Just a note, not to be solved now - but we should clean up this whole function so that it does not use getJobs() to load all jobs into memory at once - it is unnecessary - unless caching is enabled.